pkg/controller/config_test.go

```diff
@@ -0,0 +1,246 @@
+package controller
+
+import (
+	"flag"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestParseDeletionProtection(t *testing.T) {
+	t.Run("should use default values when flags or env vars were not set", func(t *testing.T) {
```

Comment:
env vars and flags test belong to main_test IMHO

here we would test config internal logic if any, maybe thing like `ParseDeletionProtection` although without flags it would be more like `SetupDeletionProtection`.